### ISSUE TITLE
Assertion triggerred in SMP+MCS kernel on HiFive Unleashed

### ISSUE THREAD

**Issue by Ivan-Velickovic** [seL4/seL4#1053]
The following assertion has been seen to non-deterministically trigger on the HiFive Unleashed platform:
```
seL4 failed assertion 't0 <= margin + t && t <= t0 + margin' at /github/workspace/kernel/src/kernel/boot.c:563 in function clock_sync_test
```

This has been seen on the debug SMP+MCS kernels. An example of the assertion being triggered can be found here https://github.com/seL4/seL4/actions/runs/5141337447/jobs/9276716395.

The full logs with the assertion failing for `HIFIVE_debug_SMP_MCS_gcc_64` can be found here https://gist.github.com/Ivan-Velickovic/c6a08044ed68228edd0b54533b8f9bad.

**Comment by Ivan-Velickovic** [comment]
The following command can be used to build the sel4test image: `../init-build.sh -DRISCV64=TRUE -DPLATFORM=hifive -DBAMBOO=TRUE -DSMP=TRUE -DMCS=TRUE`

**Comment by Indanz** [comment]
The line before the assertion is:

> clock_sync_test[1]: t0 = 23325675, t = 23325678, td = 3

The timer frequency for the HiFive is 1 MHz, so this means that the time difference between two cores is 3 us. When adding this test I already added an arbitrary 1 us margin for the HiFive, but apparently that's not enough.

To rule out a bug in `clock_sync_test()`, does replacing `__atomic_thread_fence(__ATOMIC_ACQUIRE)` with `__atomic_thread_fence(__ATOMIC_ACQ_REL)` help? That might prevent reordering `ksNumCPUs++` with the clock sync test, but I don't expect it to help.

I suspect the real fix is solving #942, which would improve RISCV performance greatly.

**Comment by Ivan-Velickovic** [comment]
Since closing https://github.com/seL4/seL4/issues/942 I have not observed this failure anymore. If the assertion is triggered again we can re-open this issue.
